### PR TITLE
Issue/91/implement language setting

### LINKED ISSUE #91: Implement setting to change language
We now have more than one languages in our app but no language selection button. This needs to be implemented

### PR DESCRIPTION
**Pull request by spuday90**
closes #91

### CODE REVIEW

**Comment by spuday90** [comment]
Note: Need to implement language switch logic

File: app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.parishod.watomatic.fragment;
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java:1]: @spuday90 We might need a separate activity for Settings.

New settings activity or new settings activity+fragment is what we need.

Or how about this:
Android Studio: `File` ➔ `New` ➔ `Activity` ➔ `Settings Activity`
And trust Android studio will recommend the best.

Comment (adeekshith) [reply app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java:1]: I need to do some research to validate/refute above suggestion but this intuitively feels wrong.

Yes, the entire app can have a single activity and a hundred fragments but what my intuitive(without much reading) suggestion is:
1. Use separate activity wherever you can. Typically every screen (in general sense) can be an activity.
2. Split Activity into fragments wherever it makes sense.
3. (if anything better exists in Android architecture components, go for it)

In our case, according to rule 1, settings can be put in a separate activity so we put it in a separate activity.

**Review by adeekshith — CHANGES_REQUESTED** [review]
- Implement language settings functionality
- Move Settings to separate class
- Fix below bug happening when dark/light themes are switched:

Screen Shot 2021-02-28 at 10 24 07 PM

File: app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    private ListPreference languagePref;
+
+    @Override
+    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
+        setPreferencesFromResource(R.xml.fragment_settings, rootKey);
+
+        //Set fragment title
+        getActivity().setTitle("Settings");
+
+        ((MainActivity)getActivity()).showHideBackButton(true);
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java:24]: We can also avoid these hacks if we move to separate activity if my understanding is right.

Comment (adeekshith) [reply app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java:24]: Or I should ask, what does this do? I see this in MainFragment too

**Comment by adeekshith** [comment]
@spuday90 merged main to this branch so you may need to pull it first to avoid conflicts

**Comment by spuday90** [comment]
>     * Implement language settings functionality
> 
>     * Move Settings to separate class
> 
>     * Fix below bug happening when dark/light themes are switched:
> 
> 
> Screen Shot 2021-02-28 at 10 24 07 PM

Looks like this is fixed with separate activity for settings, Verify on your device and let me know

**Comment by adeekshith** [comment]
Lots of changes not just adding activity and I do not understand some of them so I might take some sweet time reviewing it 😃.

File: app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+                    //Should restart the app for language change to take into account
+                    restartApp();
+                }
+                return true;
+            }
+        });
+    }
+
+    private void restartApp() {
+        Intent intent = new Intent(getActivity(), MainActivity.class);
```
Comment (adeekshith) [inline app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java:36]: This should restart and open Settings activity not MainActivity. Small change but may be we can file an issue and do it later too.

Comment (adeekshith) [reply app/src/main/java/com/parishod/watomatic/fragment/SettingsFragment.java:36]: But great work @spuday90 😃👌

**Review by adeekshith — APPROVED** [review]
Made minor changes:
- Change prefs key
- Fix "Settings" menu label

**Pull request merged**